src/utils.ts

```diff
@@ -0,0 +1,103 @@
+import { ethers, Contract, BigNumber } from "ethers";
+import { Finding, FindingSeverity, FindingType, Alert } from "forta-agent";
+import {
+  L2_FUNCTION_SIGNATURE,
+  L2_TOKEN_ADDRESS_MAKER_DAO,
+  previousBalances,
+} from "./constants";
+
+export const getL1Finding = async (
+  daiContract: Contract,
+  blockNumber: number,
+  escrowAddressArbitrum: string,
+  escrowAddressOptimism: string,
+  previousBalances: previousBalances,
+) => {
+  const arbitrumBalance: BigNumber = await daiContract.balanceOf(
+    escrowAddressArbitrum,
+    {
+      blockTag: blockNumber,
+    },
+  );
+  const optimismBalance: BigNumber = await daiContract.balanceOf(
+    escrowAddressOptimism,
+    {
+      blockTag: blockNumber,
+    },
+  );
+
+  if (
+    previousBalances.prevArbitrumBalance != arbitrumBalance &&
+    previousBalances.prevOptimismBalance != optimismBalance
+  ) {
+    previousBalances.prevArbitrumBalance = arbitrumBalance;
+    previousBalances.prevOptimismBalance = optimismBalance;
+    previousBalances.previousBlockNumber = blockNumber;
+
+    return Finding.fromObject({
+      name: `Combined supply of optimism and Arbitrum MakerDao escrows on layer 1`,
```

Comment:
Suggestion for a better term
```suggestion
      name: `Combined DAI balance of Optimism and Arbitrum MakerDao escrows on layer 1`,
```